### PR TITLE
Allow optimistic assignee chat reports

### PR DESCRIPTION
**Pull request by thienlnam**
… assignee
cc @jasperhuangg 
### Fixed Issues

$ https://github.com/Expensify/App/issues/24546
$ https://github.com/Expensify/Expensify/issues/294203
PROPOSAL: 

### Tests

1. Create a task
2. Assign someone you have not talked to before (but is still a valid account)
3. Verify it creates successfully
4. Verify the chat report between you and the assignee is also created successfully and has the correct message in the LHN
5. Edit the assignee and choose someone you have not chatted with before. Verify that another chat report is created between you and the assignee 

- [ ] Verify that no errors appear in the JS console

### Offline tests

Same as online tests

### QA Steps

1. Create a task
2. Assign someone you have not talked to before (but is still a valid account)
3. Verify it creates successfully
4. Verify the chat report between you and the assignee is also created successfully and has the correct message in the LHN

1. Create a task
2. Choose to share in that same chat report, but choose a different assignee that you haven't chatted with before
3. Verify that all the reports are created successfully
The task report
The chat report between you and the assignee
And the share somewhere report is updated correctly

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/30609178/ffd12d9b-1def-4a27-8a48-73a3b5fabda3

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/30609178/1c0418a3-ed79-4c01-abbb-16a078c705d3

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/30609178/889799f6-e2cb-4910-87dd-1d407ccac0b1

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/30609178/833cdca5-207c-4b77-a2db-74af427ad1da

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by thienlnam** [comment]
Been testing some changes and realized some additional BE changes are required https://github.com/Expensify/Web-Expensify/pull/38300

**Comment by thienlnam** [comment]
This is good to be tested on staging!

**Comment by thienlnam** [comment]
Okay linked PR is on production - this is good to go through a full testing

Thanks! cc @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
Thanks @thienlnam Quick question

So the new assignee should have a valid account for both steps 2 & 5 right?

**Comment by thienlnam** [comment]
Yeah that's correct - it has to be a valid account (exists somewhere) otherwise it will fail

**Comment by Santhosh-Sellavel** [comment]
Tests pass,
There is glitchy behavior (got navigated to concierge) while switching to the newly edited assignee's chat, observe the last 20 secs. Also, the LHN of the same chat does not reflect correct details for some time even though the assignee name is updated.

https://github.com/Expensify/App/assets/85645967/8f904e6b-e948-4747-9001-e5b18a0ae6d9

**Comment by Santhosh-Sellavel** [comment]
Bug: Also here, after changing the assignee, if you navigate to the newly created chat and click the header to see details. Email is missing even though I just now used the email to assign the user to a task.


https://github.com/Expensify/App/assets/85645967/76c0e75e-485c-470b-8443-3e86b9582ea2

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/libs/actions/Task.js [hunk 1]
```diff
@@ -33,53 +38,51 @@ function clearOutTaskInfo() {
...
+ * 3a. The CreatedReportAction for the assignee chat report
+ * 3b. The TaskReportAction on the assignee chat report
+ *
  * @param {String} parentReportID
  * @param {String} title
  * @param {String} description
  * @param {String} assignee
  * @param {Number} assigneeAccountID
- *
+ * @param {Object} assigneeChatReport - The chat report between you and the assignee
```
[the remark this answers is not in the document]
Comment (thienlnam) [reply src/libs/actions/Task.js]: What do you mean here, updating Object to something else?

File: src/libs/actions/Task.js [hunk 2]
```diff
@@ -359,8 +357,9 @@ function reopenTask(taskReportID, taskTitle) {
  * @param {String} editedTask.description
  * @param {String} editedTask.assignee
  * @param {Number} editedTask.assigneeAccountID
+ * @param {Object} assigneeChatReport - The chat report between you and the assignee
  */
```
Comment (Santhosh-Sellavel) [inline src/libs/actions/Task.js:361]: Check the other params `editedTask` is not needed here

**Comment by thienlnam** [comment]
> Bug: Also here, after changing the assignee, if you navigate to the newly created chat and click the header to see details. Email is missing even though I just now used the email to assign the user to a task.

I think this might be expected or unrelated - the same happens if you create a new chat with a user through the new chat flow. If you click the header it doesn't show their email

**Comment by thienlnam** [comment]
> There is glitchy behavior (got navigated to concierge) while switching to the newly edited assignee's chat, observe the last 20 secs. Also, the LHN of the same chat does not reflect correct details for some time even though the assignee name is updated.

Hmm yeah it appears that it does not recognize that you have an existing chat with the user and so it tries to duplicate it, but then realizes you already have one. Looking into why this happens

**Comment by thienlnam** [comment]
Okay, resolved the bug that you mentioned regarding the 
> There is glitchy behavior (got navigated to concierge) while switching to the newly edited assignee's chat

The issue lies in how we handle personal details right now. Basically 
- You create a new chat with an email
- The chat is created, but we don’t return the login so now in the personal details we just have your accountID, and display name
- You don’t show up in the selector because you haven’t chatted back, so I try to assign a task to that same email, but it doesn’t know that it’s the same user that I chatted with before and so it generates a different optimistic accountID for your account in my personal details

By checking for the existing report using the email, we can get around this issue but it creates the known bug below.

Screenshot 2023-07-31 at 3 28 19 PM

Updating that will take some more back-end changes which I'd like to do in a separate PR so this doesn't run into a bunch of merge conflicts as it also cleans up a lot of task code for anyone else trying to work in it

File: src/libs/actions/Task.js (hunk 2 above)
Comment (Santhosh-Sellavel) [reply src/libs/actions/Task.js:361]: bump on this

Comment (thienlnam) [reply src/libs/actions/Task.js]: Removed

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by Santhosh-Sellavel** [comment]
Assigning the same user another task assignee is broken.

#### Before https://github.com/Expensify/App/pull/23773
https://github.com/Expensify/App/assets/85645967/3d334e39-6144-4230-9b07-7c705e26ebcf

#### After https://github.com/Expensify/App/pull/23773
https://github.com/Expensify/App/assets/85645967/db1b8bc2-2efc-4b86-93cb-267825e0f37f

**Comment by Santhosh-Sellavel** [comment]
After editing assignee as other new users, LHN shows two chat for the same user
https://github.com/Expensify/App/assets/85645967/e12160cf-15e7-4c7a-9b9e-f3a93a830ab0

**Comment by thienlnam** [comment]
Appears to be a bug on the BE since I can't get it to happen offline - looking into it

**Comment by thienlnam** [comment]
Found the fix - PR up for the BE https://github.com/Expensify/Auth/pull/8427

**Comment by thienlnam** [comment]
BE fix is live on production - should be good now cc @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
The edited assignee is not updated in the chat which is shared the first time.

https://github.com/Expensify/App/assets/85645967/2bfbc351-1969-473a-af35-f522acf377d6

**Comment by thienlnam** [comment]
More BE changes required

**Comment by thienlnam** [comment]
I'm starting to use EditTaskAssignee, but it will be cleaned up by @Beamanator in this internal issue https://github.com/Expensify/Expensify/issues/299956

**Comment by thienlnam** [comment]
Okay, backend PR is on production - this is ready to be reviewed again @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
On it now

**Comment by Santhosh-Sellavel** [comment]
#### Online
https://github.com/Expensify/App/assets/85645967/31fda7a0-8bfc-4ab5-a36a-f41450db2766

#### Offline
##### While Offline
https://github.com/Expensify/App/assets/85645967/053a0439-7e11-48f4-9ecf-0320b122cdca

##### Back online
https://github.com/Expensify/App/assets/85645967/c77f5ffd-dab5-4356-a8e8-72667b58f5a7

**Comment by Santhosh-Sellavel** [comment]
@thienlnam ATM Tests pass, I see [here](https://github.com/Expensify/App/pull/22778#issuecomment-1672409187) the newly created chats get hidden for 1 - 2 seconds and reappear immediately. The same happens in offline cases too after coming offline IMO That's a little more odd comparing online.

The above issue not happening in the offline flow for the new chat report creation flow. Check the video below. What we can do should it be handled here or can we create it as a follow-up LMK thanks!

https://github.com/Expensify/App/assets/85645967/5809a50c-6c0c-40ea-852a-85a4745bb6dd

**Comment by thienlnam** [comment]
Let's just handle it as a follow up and try to get this PR out soon. I'll create an issue for it

**Comment by jeet-dhandha** [comment]
I noticed one thing in PR upon testing. 
1. If i create a task on a fresh account then in LHN `[ CONCIERGE_CHAT, TASK_1, CHAT_WITH_TASK_1_ASSIGNEE ]` these three reports are visible.
2. Now if i create a task again in LHN `[ CONCIERGE_CHAT, TASK_2, TASK_1, CHAT_WITH_TASK_1_ASSIGNEE ]` these four reports are visible.

Is this correct ?

Logs of LHN Report Ids:
```
1. Fresh Account: 
['1358054211663491']

2. Created a task:
['1358054211663491', '6436653505391563', '7921716444336778']

3. Created a task:
['1358054211663491', '1285478335508458', '6436653505391563', '7921716444336778'] - First changed to this
['1358054211663491', '1285478335508458', '7875153698109942', '6436653505391563', '7921716444336778'] - Then changed to this
['1358054211663491', '7875153698109942', '6436653505391563', '7921716444336778'] - Then again changed to this
['1358054211663491', '6436653505391563', '7921716444336778'] - Then again changed to this
['1358054211663491', '1285478335508458', '6436653505391563', '7921716444336778'] - finally changed to this
```

**Comment by thienlnam** [comment]
Yes that's correct, and how it should work today

**Comment by jeet-dhandha** [comment]
I was just concerned about other chats not getting created when a task was created like it did in fresh account.

**Comment by jeet-dhandha** [comment]
Screenshot 2023-08-11 at 1 10 31 AM

Also in a failed task on press of top-left avatar `/participants` path is opened instead of `/a/15094372` (which is a profile detail's page)

**Comment by Santhosh-Sellavel** [comment]
@thienlnam I'm getting this console error
Screenshot 2023-08-11 at 1 06 46 AM

**Comment by jeet-dhandha** [comment]
Also my issue can be fixed by adding this. (https://github.com/Expensify/App/pull/22778#issuecomment-1673808942)
1. We basically need to add participant ids as the `managerId` but on front end i don't think we have it so instead i thought of adding `currentUserAccountID` which also works but issue might happen as we move further.

```diff
function createTaskAndNavigate(parentReportID, title, description, assigneeEmail, assigneeAccountID = 0, assigneeChatReport = null) {
    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserAccountID, assigneeAccountID, parentReportID, title, description);

+   optimisticTaskReport.participantAccountIDs = [currentUserAccountID];
```

**Comment by thienlnam** [comment]
I haven't been able to reproduce the console error after merging main on iOS or mWeb but I'm not sure since my android emulator isn't working.

Also, it's fine to not worry about a failed task right now

**Comment by Santhosh-Sellavel** [comment]
[This](https://github.com/Expensify/App/pull/22778#issuecomment-1673812212) occurs on the main too, so its not blocking this.

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM, thanks!

**Comment by thienlnam** [comment]
Wooo!! Cc @jasperhuangg we're ready to go here

**Review by jasperhuangg — APPROVED** [review]
nicely done! tests well and don't have any qualms with the code

**Comment by ayazhussain79** [comment]
@thienlnam @jasperhuangg Unable to select "Assignee" after creating task due to console warning on IOS
https://expensify.slack.com/archives/C049HHMV9SM/p1691795662145199

**Comment by thienlnam** [comment]
Hmm yeah, apparently when first created the report is empty so it doesn't work
But if you exit and go back to the task you can re-assign it fine

**Comment by michaelhaxhiu** [comment]
👋 is the bug report https://github.com/Expensify/App/issues/25956 a regression from this PR? Curious if we should fix it separately (on the GH linked above) or re-open the parent GH/PR here that this PR touched?

**Comment by Santhosh-Sellavel** [comment]
I guess follow-up would be the case, cc: @thienlnam can help here!

File: src/libs/actions/Task.js [hunk 3]
```diff
@@ -578,40 +629,65 @@ function setShareDestinationValue(shareDestination) {
...
+                accountID: assigneeAccountID,
+                avatar: lodashGet(allPersonalDetails, [assigneeAccountID, 'avatar'], UserUtils.getDefaultAvatarURL(assigneeAccountID)),
+                displayName: lodashGet(allPersonalDetails, [assigneeAccountID, 'displayName'], assigneeEmail),
+                login: assigneeEmail,
+            };
+            Onyx.merge(ONYXKEYS.PERSONAL_DETAILS_LIST, {[assigneeAccountID]: optimisticPersonalDetailsListAction});
         }
-        const reportID = chat ? chat.reportID : newChat.reportID;
 
+        setAssigneeChatReport(chatReport);
```
Comment (hungvu193) [inline src/libs/actions/Task.js:675]: This line introduces a bug [here](https://github.com/Expensify/App/issues/38439) where the user chose a brand new assignee (whom they've never chatted with before), and then chose another old assignee, leading to the incorrect merging of `isOptimisticReport` into the old assignee's account.

**Pull request merged**